### PR TITLE
Native event emitter support

### PR DESCRIPTION
**Pull request by chatrapathi06**
## Description
Native Event emitter support is added to RNS. 

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| Linux   |    ✅     |
| iOS     |    ❌     |
| Android |    ❌     |
| Windows |    ❌     |

## Checklist

* [x] I have tested this on Ubuntu OS
* [ ] I added the documentation in `README.md`
* [ ] I updated the typings files (`privateTypes.ts`, `types.ts`)
* [ ] I added a sample use of the API (`example/App.js`)

### CODE REVIEW

File: ReactSkia/LegacyNativeModules/NativeEventEmitter.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+          listenerCount_++;
+          if (listenerCount_ == 1) {
+            startObserving();
+          }
+        }),// end of addListener lambda
+
+    Method(
+        "removeListeners",
+        [&] (folly::dynamic args) {
+          int  removeCount = args[0].asInt();;
```
Comment (shefayaparwin) [inline ReactSkia/LegacyNativeModules/NativeEventEmitter.cpp]: double semicolon

Comment (shefayaparwin) [reply ReactSkia/LegacyNativeModules/NativeEventEmitter.cpp]: Does the removeListener method consider the number of listeners to remove? If so, what is the use case for it? Does this mean that removing multiple listeners will be done in batches when calling the removeListener method?

File: ReactSkia/LegacyNativeModules/NativeEventEmitter.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          }
+        }),
+  };
+}
+
+void NativeEventEmitter::sendEventWithName(std::string eventName, folly::dynamic eventData) {
+  auto instance = getInstance().lock();
+  if ( instance ) {
+    instance->callJSFunction(
+            "RCTDeviceEventEmitter", "emit",
```
Comment (shefayaparwin) [inline ReactSkia/LegacyNativeModules/NativeEventEmitter.cpp]: Is it specific to RCTDeviceEventEmitter module, or generic base class

File: ReactSkia/LegacyNativeModules/NativeEventEmitter.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        [&] (folly::dynamic args) {
+          removeListeners(args[0].asInt());
+        }),
+  };
+}
+
+void NativeEventEmitter::sendEventWithName(std::string eventName, folly::dynamic &&params, EmitterCompleteVoidCallback completeCallback) {
+  auto instance = getInstance().lock();
+  if(instance){
+    SetBridgeInstance(instance.get());
```
Comment (chatrapathi06) [inline ReactSkia/LegacyNativeModules/NativeEventEmitter.cpp:42]: if instance is created don't set the instance

**Review by munezbn — APPROVED** [review]

File: ReactSkia/LegacyNativeModules/NativeEventEmitter.h [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+#pragma once
+#include <cxxreact/CxxModule.h>
+#include "ReactSkia/LegacyNativeModules/RSkBaseEventEmitter.h"
+#include "ReactSkia/utils/RnsUtils.h"
+
+
+namespace facebook {
+namespace xplat {
+using namespace react;
+class NativeEventEmitter : public module::CxxModule, public react::RSkBaseEventEmitter{
```
Comment (munezbn) [inline ReactSkia/LegacyNativeModules/NativeEventEmitter.h]: I thin we shoold name calss as NativeEventEmitterModule

**Review by shefayaparwin — APPROVED** [review]

**Pull request merged**